pumpkin-lib/src/propagators/reified_propagator.rs

```diff
@@ -180,12 +183,23 @@ impl<Prop> ReifiedPropagator<Prop> {
         Ok(())
     }
 
+    fn find_inconsistency(&mut self, context: PropagationContext<'_>) -> bool {
+        if self.inconsistency.is_none() {
+            self.inconsistency = self.propagator.detect_inconsistency(context);
+        }
+
+        self.inconsistency.is_some()
+    }
+
     fn filter_enqueue_decision(
-        &self,
+        &mut self,
         context: PropagationContext<'_>,
         decision: EnqueueDecision,
     ) -> EnqueueDecision {
-        if decision == EnqueueDecision::Enqueue && context.is_literal_true(self.reification_literal)
+        if decision == EnqueueDecision::Enqueue
```

Comment:
This is really difficult to read and understand. It can be broken up by early returns, or simply creating local boolean variables with descriptive names.